pkg/build/controller/factory/factory.go

```diff
@@ -96,9 +96,13 @@ func (factory *BuildControllerFactory) Create() controller.RunnableController {
 			build := obj.(*buildapi.Build)
 			err := buildController.HandleBuild(build)
 			if err != nil {
-				build.Status.Message = err.Error()
-				if err := buildController.BuildUpdater.Update(build.Namespace, build); err != nil {
-					glog.V(2).Infof("Failed to update status message of Build %s/%s: %v", build.Namespace, build.Name, err)
+				msg := err.Error()
+				// Update the build status message only if it changed.
+				if build.Status.Message != msg {
+					build.Status.Message = msg
+					if err := buildController.BuildUpdater.Update(build.Namespace, build); err != nil {
+						glog.V(2).Infof("Failed to update status message of Build %s/%s: %v", build.Namespace, build.Name, err)
+					}
```

Comment:
is this the only place we'll ever update the status.message?  wondering if this logic belongs in a utility function.